Microsoft.Research/Contracts/MsCorlib/System.Collections.Generic.IDictionary.cs

```diff
@@ -181,29 +183,31 @@ abstract class IDictionaryContract<TKey, TValue> : IDictionary<TKey,TValue>
 
     void IDictionary<TKey, TValue>.Add(TKey key, TValue value)
     {
+      Contract.Requires(!ReferenceEquals(key, null));
       // Contract.Requires(!ContainsKey(key));
       //modifies this.*;
       //Contract.Ensures(ContainsKey(key));
     }
 
-    bool IDictionary<TKey, TValue>.ContainsKey(TKey key)
+    public bool ContainsKey(TKey key)
```

Comment:
:thought_balloon: I find the inconsistency between explicitly and implicitly implemented methods a bit confusing. Note that in addition to understanding why you made the change, I don't have a preference towards implicit or explicit implementations of contract methods overall (each has their merits, and both work).